### ISSUE TITLE
Shouldn't we better shuffle data for pretraining on document level instead of sentence level?

### ISSUE THREAD

**Issue by tstadel** [deepset-ai/FARM#426]
**Question**
`farm.data_handler.utils.randomize_and_split_file()` shuffles training data on sentence / line level. This not only introduces many warnings since empty lines get concatenated by chance thus producing empty documents. But it also breaks NextSentencePrediction learning tasks as they need correct sentence sequences. Even if NextSentencePrediction may be widely regarded as obsolete, I guess, we shouldn't destroy it implicitly...

So shouldn't we better shuffle on document level instead of sentence level?

**Additional context**
`farm.data_handler.utils.randomize_and_split_file()` is used in examples/train_from_scratch.py

**Comment by tanaysoni** [comment]
Hi @tstadel, thank you for raising the issue. @tholor and I discussed potential solutions to resolve this. A quick fix with #427 is to refactor the `randomize_and_split_file()` to `split_file()`, where a large train file is split in smaller files each having a document count of `docs_per_file`. 

For the randomization, we can rely on `farm.data_handler.data_silo._StreamingDataSet.shuffle_files()` to shuffle the order of reading the split files during each training epoch. The `docs_per_file` param should be reasonably small for this approach to work well.

If we don't get good results with this approach, we can perhaps create a separate PR to implement an additional `randomize_file()` that can be called before each training epoch to randomize the order of docs in individual split files.

What do you think?

**Comment by tstadel** [comment]
Hi @tanaysoni,

looks appropriate to me.
